AmplifyPlugins/Core/AWSPluginsCore/Model/GraphQLDocument/SingleDirectiveGraphQLDocument.swift

```diff
@@ -0,0 +1,111 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import Foundation
+
+public enum GraphQLOperationType: String {
+    case mutation
+    case query
+    case subscription
+}
+
+public typealias GraphQLParameterName = String
+
+/// Represents a single directive GraphQL document. Concrete types that conform to this protocol must
+/// define a valid GraphQL operation document.
+///
+/// This type aims to provide a representation of a simple GraphQL document with its components that can be decorated
+/// and later serialized into query document and variables for a request to a GraphQL service.
+/// Therefore, documents represented by concrete implementations provide a single GraphQL
+/// operation based on a defined `Model.Type`.
```

Comment:
I'm missing how this protocol relies on Model. It seems generic enough to represent any single directive document, even if such is based on something other than a model. However, it might be worth a bit more explanation on the use cases we're trying to solve:

* Single directives rather than multiple directives
* Easily derivable and decoratable inputs and selection sets with standardized output forms, including standardized names
* ...